src/lib/optimizer/strategy/like_replacement_rule.cpp

```diff
@@ -0,0 +1,87 @@
+#include "like_replacement_rule.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+#include "all_parameter_variant.hpp"
+#include "constant_mappings.hpp"
+#include "expression/expression_functional.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "operators/operator_scan_predicate.hpp"
+#include "statistics/chunk_statistics/chunk_statistics.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/assert.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+std::string LikeReplacementRule::name() const { return "Like Replacement Rule"; }
+
+void LikeReplacementRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) const {
+  const auto predicate_node = std::dynamic_pointer_cast<PredicateNode>(node);
+  // Continue only on predicate nodes
```

Comment:
I don't know if we really need this (same applies to L33) comment, line 26 and 28 are pretty easy to understand and it should be self-explanatory. But I would not block this PR if you decide to leave it like this.